services/cd-service/pkg/repository/repository.go

```diff
@@ -2097,6 +2142,52 @@ func (s *State) GetCurrentEnvironmentLocks(ctx context.Context, _ *sql.Tx) (db.A
 	return result, nil
 }
 
+func (s *State) GetCurrentApplicationLocks(ctx context.Context, tx *sql.Tx) (db.AllAppLocks, error) {
```

Comment:
```suggestion
func (s *State) GetCurrentApplicationLocks(ctx context.Context) (db.AllAppLocks, error) {
```
This should not require any database access and therefore no transaction either.